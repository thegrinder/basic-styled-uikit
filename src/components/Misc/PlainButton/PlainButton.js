import styled from 'styled-components';

const PlainButton = styled.button`
  font-family: inherit;
  font-size: 1rem;
  border: none;
  overflow: visible;
  text-transform: none;
  background-color: transparent;
  cursor: pointer;
  display: inline-block;
  position: relative;
  fill: currentcolor;
  line-height: 1;
  outline: none;
  transition: 0.1s ease-in-out;
  transition-property: color;
`;

export default PlainButton;
