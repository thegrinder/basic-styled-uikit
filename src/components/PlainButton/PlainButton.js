import styled from 'styled-components';
import { string } from 'prop-types';
import { getColor } from '../../theme/colors';

const propTypes = {
  color: string,
  hoverColor: string,
};

const defaultProps = {
  color: 'gray700',
  hoverColor: 'gray900',
};

const PlainButton = styled.button`
  touch-action: manipulation;
  margin: 0;
  border: none;
  overflow: visible;
  text-transform: none;
  padding: 0;
  background-color: transparent;
  cursor: pointer;
  display: inline-block;
  position: relative;
  fill: currentcolor;
  line-height: 0;
  outline: none;
  transition: .1s ease-in-out;
  transition-property: color;
  color: ${({ theme, color }) => getColor(theme, color)};
  &:focus,
  &:hover {
    opacity: 1;
    color: ${({ theme, hoverColor }) => getColor(theme, hoverColor)};
  }
`;

PlainButton.propTypes = propTypes;
PlainButton.defaultProps = defaultProps;

export default PlainButton;
