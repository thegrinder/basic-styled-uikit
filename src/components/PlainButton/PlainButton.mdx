---
name: PlainButton
menu: Components
---

import { Playground } from 'docz';
import PlainButton from './PlainButton';

# PlainButton

<Playground>
  <PlainButton>plain button</PlainButton>
</Playground>
