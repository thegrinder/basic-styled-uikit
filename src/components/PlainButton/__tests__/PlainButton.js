import React from 'react';
import { render, cleanup } from '@testing-library/react';


import PlainButton from '../PlainButton';

describe('<PlainButton />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { container: { firstChild } } = render(
      <PlainButton />,
    );
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <button> tag', () => {
    const { container: { firstChild } } = render(
      <PlainButton />,
    );
    expect(firstChild.tagName).toEqual('BUTTON');
  });

  it('should render text correctly', () => {
    const text = 'text';
    const { container: { firstChild } } = render(
      <PlainButton>
        {text}
      </PlainButton>,
    );
    expect(firstChild).toHaveTextContent(text);
    expect(firstChild).toMatchSnapshot();
  });
});
