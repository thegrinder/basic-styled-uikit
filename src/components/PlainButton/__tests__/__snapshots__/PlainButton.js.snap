// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainButton /> should render text correctly 1`] = `
<button
  class="PlainButton-e9bw3w-0 kTxkaK"
>
  text
</button>
`;
